### PR TITLE
devops: Add GHA workflow for triggering semgrep-comparison Argo Workflow

### PR DESCRIPTION
**Pull request by minusworld**
This PR introduces a workflow file for triggering the `semgrep-comparison` Argo Workflow. It has the following properties, which can be amended if desired:
* When triggered manually, it will compare `develop` to `latest`.
* When triggered from a PR, it will compare `pr-NNNN` to `develop`.
* The check run will time out after 1 hour.
* The Argo Workflow will run the smallest set of repos (the `debug` setting). Repos could be configured to be dynamic.
* The Argo Workflow uses `p/default-v2`. This can be changed, or set up to be dynamic.
* The check run will report results to the check run page or a timeout error if timed out.

**Note: this workflow will require the ARGO_WORKFLOWS_TOKEN secret to be added.**

# Test plan:
## 1. Test triggering the workflow URL.
```
# login to AWS, e.g., with Leapp or okta-aws-cli
> okta-aws-cli ...
# set dev2 as kubernetes environment, e.g., with ckc dev2
> ckc dev2
# Trigger the Argo Workflow on a test repository with the Argo Workflow URL
> curl -X POST https://argoworkflows-dev2.corp.r2c.dev/api/v1/events/security-research/semgrep-compare \
    -H "Authorization: Bearer $(kubectl get secret -n security-research event-service-account-token -oyaml | yq '.data.token' | base64 -D)" \
    -d '{"repository": "returntocorp/WebGoat", "sha": "78d5bf174b43296f79d33ea26398f39cf681bffc", "ruleset": "p/default-v2", "container_image_base": "returntocorp/semgrep:latest", "container_image_development": "returntocorp/semgrep:develop"}'
```
image

## 2. Test the workflow with `act`.
```
# login to AWS, e.g., with Leapp or okta-aws-cli
> okta-aws-cli ...
# set dev2 as kubernetes environment, e.g., with ckc dev2
> ckc dev2
# Test the GHA workflow with `act`, using a `workflow_dispatch` event
> act workflow_dispatch \
    -s ARGO_WORKFLOWS_TOKEN=$(kubectl get secrets -n security-research event-service-account-token -oyaml | yq '.data.token' | base64 -D) \
    -W .github/workflows/semgrep-compare-argo-workflow.yaml \
    --input branch="gh/dagoba-fcc928" \
    -P ubuntu-22.04=catthehacker/ubuntu:act-22.04
```
image

### CODE REVIEW

**Comment by minusworld** [comment]
Results look like this.
## Success
image
## Timed out
Note: this timed out after 8+ hours due to an issue which has now been fixed.
image
## Failure
image

File: .github/workflows/semgrep-compare-argo-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: trigger-semgrep-comparison-argo-workflow
```
Comment (aryx) [inline .github/workflows/semgrep-compare-argo-workflow.yaml]: please use the same name than the file here, so when it fails it's easier to find the offending workflow

Comment (aryx) [reply .github/workflows/semgrep-compare-argo-workflow.yaml]: Can you add a toplevel comment in this explaining quickly what this workflow is for, and vaguely how it works.

File: .github/workflows/semgrep-compare-argo-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            echo "docker-tag=develop" >> "$GITHUB_OUTPUT"
+            echo "Setting dry-run to develop"
+          fi
+  get-sha:
+    name: Get SHA for attaching Argo Workflow check
+    runs-on: ubuntu-22.04
+    outputs:
+      sha: ${{ steps.get-sha.outputs.sha }}
+    steps:
+      - name: Checkout
```
Comment (aryx) [inline .github/workflows/semgrep-compare-argo-workflow.yaml]: please remove those name: Xxx when they don't add any information.

File: .github/workflows/semgrep-compare-argo-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          echo "Branch: $BRANCH"
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "sha=${{ github.event.pull_request.head.sha }}" >> "$GITHUB_OUTPUT"
+          elif [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
+            git fetch origin $BRANCH
+            SHA=$(git rev-parse $BRANCH)
+            echo "sha=$SHA" >> "$GITHUB_OUTPUT"
+          fi
+  trigger-semgrep-comparison-argo-workflow:
+    name: Trigger semgrep-comparison Argo Workflow
```
Comment (aryx) [inline .github/workflows/semgrep-compare-argo-workflow.yaml]: you can remove the name:, the name of the job contains already the info

**Review by cgdolan — APPROVED** [review]

**Review by apt-itude — APPROVED** [review]

File: .github/workflows/trigger-semgrep-comparison-argo.yaml [hunk 4]
```diff
@@ -1,4 +1,7 @@
-name: trigger-semgrep-comparison-argo-workflow
+# This workflow triggers the `semgrep-comparison` Argo Workflow, which will
+# create a GitHub Check on this pull request (or specified branch if running manually.
```
Comment (aryx) [inline .github/workflows/trigger-semgrep-comparison-argo.yaml:2]: missing ), and I don't see any on: pull_request, so I don't think this run on a pull_request.

Comment (minusworld) [reply .github/workflows/trigger-semgrep-comparison-argo.yaml:2]: This workflow will be called from the `tests` workflow, which runs on pull requests, but that will come in a forthcoming PR.

**Pull request merged**